hare/eligibility/oracle.go

```diff
@@ -270,26 +269,26 @@ func (o *Oracle) prepareEligibilityCheck(ctx context.Context, layer types.LayerI
 	)
 
 	// ensure miner weight fits in int
-	n := int(minerWeight)
-	if uint64(n) != minerWeight { // TODO(mafa): why not [if minerWeight > maxSupportedN]? leave n uint64 and cast to int (and check for overflow) before returning?
+	n := minerWeight
+	if n > maxSupportedN {
 		logger.Fatal(fmt.Sprintf("minerWeight overflows int (%d)", minerWeight))
```

Comment:
why are we exiting here but not further below when we are checking for an overflow of `n` again?

Reply:
You might be right, but just trying to nitpick here: what if you get a higher `n` and this results in an overflow (wrap around) such that the next check for a value of `n` that's over the threshold doesn't trigger?

It sounds like we shouldn't care too much about this case, because it requires crazy weight and the only thing it allows is to get fewer eligibilities, but, you know, my dad always told me "Ordnung muß sein"...

Reply:
OTOH, here we only checked that `minerWeight` fits in an `int`, so it's actually safer the way you did it. I don't think these nuances are terribly important, so I support removing this check here and keeping your other changes.

Reply:
I know that the second check is necessary - I was just nitpicking the different reactions of the node if the check doesn't pass. Shutting down seems extreme here 😅. Its probably very unlikely but if some node manages to get enough weight to pass the threshold this check will shutdown the whole network (or at least all nodes using `go-spacemesh`).